Data/Repositories/CandidateRepository.cs

```diff
@@ -9,46 +9,15 @@ public CandidateRepository(DataContext context) : base(context)
 
         }
 
-        public Candidate GetCandidateById(int id) {
-            Candidate result = GetById(id);
-
-            if (result == default) return null;
-
-            return result;
-        }
-
-        public IEnumerable<Candidate> GetAllCandidates() 
+        public Candidate GetCandidateById(int id) 
```

Comment:
You already have this logic in the base repository. There's no need to create method to use it